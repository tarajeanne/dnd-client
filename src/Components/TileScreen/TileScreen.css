* {
  box-sizing: border-box;
}

.tile-area {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
}

.Loader {
  position: fixed;
  left: 50%;
  top: 50%;
}