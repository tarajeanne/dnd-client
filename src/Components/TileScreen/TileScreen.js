import React from 'react';
import Tile from '../Tile/Tile';
import CharacterApiService from '../../services/character-api-service';
import CharacterContext from '../../contexts/CharacterContext';
import './TileScreen.css';

class TileScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      screen: this.props.match.path.split('/')[3],
      id: this.props.match.path.split('/')[2]
    };
  }

  static contextType = CharacterContext;

  handleClick = (name) => {
    CharacterApiService.updateCharacter(
      this.state.id,
      this.state.screen,
      name
    ).then((res) => {
      this.context.setCharacter(res);
      this.setState({
        selected: this.context.character[this.state.screen]
      });
    });
  };

  componentDidMount() {
    CharacterApiService.getLists(this.state.screen).then((res) => {
      this.setState({
        [this.state.screen]: res,
        selected: this.context.character[this.state.screen]
      });
    });
  }

  render() {
    if (this.state[this.state.screen]) {
      const allElements = this.state[this.state.screen].map((item, index) => {
        return (
          <Tile
            key={index}
            id={index + 1}
            item={item}
            handleClick={this.handleClick}
            selected={this.state.selected}
          />
        );
      });

      return <div className="tile-area">{allElements}</div>;
    } else {
      return <div>loading...</div>;
    }
  }
}

export default TileScreen;
