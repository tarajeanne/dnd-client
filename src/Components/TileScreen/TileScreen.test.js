import React from 'react';
import TileScreen from './TileScreen';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('TileScreen', () => {
    it('renders as expected', () => {
      const match = { path: 'http://localhost:3000/character/cjzc2ny7q0000fep28oy727gq/race'}
        const wrapper = shallow(<TileScreen match={match}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
  });
