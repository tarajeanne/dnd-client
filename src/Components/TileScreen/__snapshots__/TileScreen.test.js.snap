// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileScreen renders as expected 1`] = `<Fragment />`;
